pkg/features/kube_features.go

```diff
@@ -396,7 +390,6 @@ var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureS
 	RunAsGroup:                                  {Default: false, PreRelease: utilfeature.Alpha},
 	VolumeSubpath:                               {Default: true, PreRelease: utilfeature.GA},
 	BalanceAttachedNodeVolumes:                  {Default: false, PreRelease: utilfeature.Alpha},
-	DynamicProvisioningScheduling:               {Default: false, PreRelease: utilfeature.Alpha},
```

Comment:
shouldn't this change to true and beta, not be removed completely?

Reply:
We want to merge this with the VolumeScheduling feature gate, which is already beta

Reply:
According to the feature gate discussion [here](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/kubernetes-sig-architecture/qJeJ7ANYzS0/qll1H4GoDQAJ), when a flag goes GA, we have to deprecate it before removing, to support upgrade cases.  

However, this is an alpha feature gate, so we don't support upgrade.  I think it is fine to remove/merge it in this case.  @liggitt wdyt?

Reply:
if the gate never graduated from alpha, we can drop it on a release boundary (though it's release-note worthy).

based on offline discussion, the code gated by DynamicProvisioningScheduling is also opt-in based on storageclass attributes. if that's the case, I see no downside to collapsing onto the VolumeScheduling

Reply:
Making sure all entry points to the feature are covered by:
1) VolumeScheduling feature gate
2) Opt-in behavior via StorageClass.VolumeBindingMode field

Scheduler:
- Feature gate check is at the beginning of the [predicate](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/predicates/predicates.go#L1675)
- In predicate call, we check for delayed binding setting and treat it as [legacy](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/scheduler_binder.go#L383) (immediate binding) scenario otherwise
- Assume/bind volumes are feature [gated](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go#L268)

PV controller:
- Defer to the scheduler is controlled by feature [gate](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/pv_controller.go#L289) and StorageClass [check](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/pv_controller.go#L317)

API server:
- StorageClass field controlled by feature [gate](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/storage/validation/validation.go#L239)

Reply:
@lichuqiang can you update the release note saying that the feature gate has been merged with VolumeScheduling.

Reply:
done

Reply:
sorry being nitpicky, can you word it as "ACTION REQUIRED: The DynamicProvisioningScheduling alpha feature gate has been removed.  The VolumeScheduling beta feature gate is still required for this feature"

Reply:
sure thing